debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/IncrementalSnapshotCollationSortOrderMismatchIT.java

```diff
@@ -0,0 +1,395 @@
+        orderMismatchPkTypecharValueInt(true, ALL_IDS.size() - SKIPPED_IDS.size(), "varchar(50) COLLATE " + SQL_COLLATION);
+    }
+
+    // Ensure unicode values are read back ok
+    @Test
+    public void orderMismatchPkVarcharValueNvarcharParamsAsUnicodeFalse() throws Exception {
+        orderMismatchPkVarcharValueNvarchar(false, ALL_IDS.size());
+    }
+
+    @Test
+    public void orderMismatchPkVarcharValueNvarcharParamsAsUnicodeTrueSkip36() throws Exception {
+        orderMismatchPkVarcharValueNvarchar(true, ALL_IDS.size() - SKIPPED_IDS.size());
+    }
+
+    //
+    // PK - NCHAR
+    @Test
+    public void orderMismatchPkNcharValueNvarcharParamsAsUnicodeFalse() throws Exception {
+        orderMismatchPkNtypeValueNvarchar(false, ALL_IDS.size(), "nchar(50)");
+    }
+
+    @Test
+    public void orderMismatchPkNcharValueNvarcharParamsAsUnicodeTrue() throws Exception {
+        orderMismatchPkNtypeValueNvarchar(true, ALL_IDS.size(), "nchar(50)");
+    }
+
+    //
+    // PK - NTEXT - Cannot be used as a PK column so not tested.
+
+    //
+    // PK - NVARCHAR
+    @Test
+    public void orderMismatchPkNvarcharValueNvarcharParamsAsUnicodeFalse() throws Exception {
+        orderMismatchPkNtypeValueNvarchar(false, ALL_IDS.size(), "nvarchar(50)");
+    }
+
+    @Test
+    public void orderMismatchPkNvarcharValueNvarcharParamsAsUnicodeTrue() throws Exception {
+        orderMismatchPkNtypeValueNvarchar(true, ALL_IDS.size(), "nvarchar(50)");
+    }
+
+    protected void orderMismatchPkTypecharValueInt(final boolean isSendStringParametersAsUnicode, int expectedRecordCount, String pkDataType) throws Exception {
+        runTest(isSendStringParametersAsUnicode,
+                expectedRecordCount,
+                pkDataType,
+                (pk, _idx) -> String.format("'%s'", pk),
+                // NOTE: we use .trim() to handle the padding added when a non-variable length type is used.
+                pk -> pk.getString(pkFieldName()).trim(),
+                "int",
+                (_pk, idx) -> String.format("%d", idx),
+                record -> ((Struct) record.value()).getStruct("after").getInt32(valueFieldName()),
+                dbChanges -> {
+                    boolean result = true;
+                    for (int i = 0; i < ALL_IDS.size(); i++) {
+                        var id = ALL_IDS.get(i);
+                        // NOTE: in the case where isSendStringParametersAsUnicode is true, the 36 records will
+                        // have been skipped (i.e. you need to set it to false for it to work).
+                        if (isSendStringParametersAsUnicode && SKIPPED_IDS.contains(id)) {
+                            continue;
+                        }
+                        var val = dbChanges.get(id);
+                        if (val == null || val != i) {
+                            result = false;
+                            System.err.println(ALL_IDS.get(i) + " value is not = " + i + ", is = " + val);
```

Comment:
```suggestion
                            Testing.Print(ALL_IDS.get(i) + " value is not = " + i + ", is = " + val);
```

Reply:
Same for the other instances

Reply:
I used Testing.printError() as I want to see the error if it occurs (it shouldn't normally). The print() method appears to need to be explicitly enabled to print something.